### ISSUE TITLE
[addon-operator] Configure TLS in AdmissionServer

### ISSUE THREAD

**Issue by meatuses** [deckhouse/deckhouse#8117]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Default TLS configuration could be insecure, e.g. supporting legacy TLS version 1.0, 1.1

Currently, it affects `deckhouse-controller`:
```
[deckhouse] deckhouse@k8s-master-0 /deckhouse $ curl https://localhost:9651 -k -v --tlsv1.0 --tls-max 1.1
* processing: https://localhost:9651
*   Trying [::1]:9651...
* Connected to localhost (::1) port 9651
* ALPN: offers h2,http/1.1
* TLSv1.1 (OUT), TLS handshake, Client hello (1):
* TLSv1.1 (IN), TLS handshake, Server hello (2):
* TLSv1.1 (IN), TLS handshake, Certificate (11):
* TLSv1.1 (IN), TLS handshake, Server key exchange (12):
* TLSv1.1 (IN), TLS handshake, Server finished (14):
* TLSv1.1 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.1 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.1 (OUT), TLS handshake, Finished (20):
* TLSv1.1 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.1 / ECDHE-ECDSA-AES128-SHA
```

### Proposed Solution

Make it possible to pass TLS configuration to addon-operator

### Additional Information

https://github.com/flant/addon-operator/blob/1c9e7b1830b55ebd508b78c96d9d11a272908c81/pkg/addon-operator/admission_http_server.go#L46-L57

**Comment by konstantin-axenov** [comment]
SSL/TLS: Deprecated TLSv1.0 and TLSv1.1 Protocol Detection (CVE-2011-3389, CVE-2015-0204).

**Comment by raabdullaev** [comment]
Fixed in Go 1.22 - #8333.
